### PR TITLE
Issue 128 master

### PR DESCRIPTION
**Pull request by ddbox**
Frontend operator adds
"""
&lt; attr name="CONTINUE_IF_NO_PROXY" glidein_publish="True" job_publish="True" parameter="True" type="string" value="True"/ &gt;
"""
to a frontend group if it has a scitoken credential only to enable routing of credentials in factory.
If parameter is missing or has value other than 'True' factory behaves as it did in 3.7.6

### CODE REVIEW

File: factory/glideFactoryCredentials.py [hunk 1]
```diff
@@ -268,7 +268,7 @@ def validate_frontend(classad, frontend_descript, pub_key_obj):
     return sym_key_obj, frontend_sec_name
 
 
-def check_security_credentials(auth_method, params, client_int_name, entry_name):
+def check_security_credentials(auth_method, params, client_int_name, entry_name, scitoken_passthru=False):
     """
     Verify taht only credentials for the given auth method are in the params
```
Comment (mambelli) [inline factory/glideFactoryCredentials.py:274]: Add also `scitoken_passthru` to the parameters and possibly convert to the google format.
```
scitoken_passthru (bool): skip the GlideinProxy requirement and allow to use only scitokens also for grid_proxy authentication method
```

**Review by mambelli — CHANGES_REQUESTED** [review]

File: factory/glideFactoryCredentials.py (hunk 1 above)
Comment (ddbox) [reply factory/glideFactoryCredentials.py:274]: done

File: factory/glideFactoryCredentials.py [hunk 2]
```diff
@@ -307,30 +307,31 @@ def check_security_credentials(auth_method, params, client_int_name, entry_name)
         "AuthFile",
     }
 
-    if "scitoken" in auth_method_list:
-        # TODO check validity
+    if "scitoken" in auth_method_list or "frontend_scitoken" in params and scitoken_passthru:
+        # TODO  check validity
```
[the remark this answers is not in the document]
Comment (ddbox) [reply factory/glideFactoryCredentials.py:311]: done

File: factory/glideFactoryCredentials.py [hunk 3]
```diff
@@ -307,30 +307,31 @@ def check_security_credentials(auth_method, params, client_int_name, entry_name)
...
-            raise CredentialError(
-                "Request from client %s did not provide a proxy as required by the entry %s, skipping request"
-                % (client_int_name, entry_name)
-            )
+            if not scitoken_passthru:
+                # No proxy sent
+                raise CredentialError(
+                    "Request from client %s did not provide a proxy as required by the entry %s, skipping request"
+                    % (client_int_name, entry_name)
+                )
```
[the remark this answers is not in the document]
Comment (ddbox) [reply factory/glideFactoryCredentials.py:329]: done

**Review by mambelli — APPROVED** [review]

**Pull request merged**